contribs/gnodev/pkg/dev/node_state_test.go

```diff
@@ -0,0 +1,189 @@
+package dev
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	emitter "github.com/gnolang/gno/contribs/gnodev/internal/mock"
+	"github.com/gnolang/gno/contribs/gnodev/pkg/events"
+	"github.com/gnolang/gno/gno.land/pkg/gnoclient"
+	"github.com/gnolang/gno/gno.land/pkg/gnoland"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNodeMovePreviousTX(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	const callInc = 5
+
+	node, emitter := testingCounterRealm(t, callInc)
+
+	t.Run("Prev TX", func(t *testing.T) {
+		err := node.MoveToPreviousTX(ctx)
+		require.NoError(t, err)
+		assert.Equal(t, events.EvtReload, emitter.NextEvent().Type())
+
+		// Check for correct render update
+		render, err := testingRenderRealm(t, node, "gno.land/r/dev/counter")
+		require.NoError(t, err)
+		require.Equal(t, render, "4")
+	})
+
+	t.Run("Next TX", func(t *testing.T) {
+		err := node.MoveToNextTX(ctx)
+		require.NoError(t, err)
+		assert.Equal(t, events.EvtReload, emitter.NextEvent().Type())
+
+		// Check for correct render update
+		render, err := testingRenderRealm(t, node, "gno.land/r/dev/counter")
```

Comment:
Can we make this realm path a constant, at least in the test? We use it a few times

Reply:
I finished reading the file, we use it a lot more than a few times